modifytags/modifytags.go

```diff
@@ -0,0 +1,367 @@
+package modifytags
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"go/ast"
+	"go/token"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode"
+
+	"github.com/fatih/camelcase"
+	"github.com/fatih/structtag"
+)
+
+// A Mutation defines how the input is modified.
+type Mutation struct {
+	Remove        []string            // Remove tags for the comma separated list of keys.
+	RemoveOptions map[string][]string // Remove the list of options per given key.
+
+	Add                  []string            // Adds tags for the comma separated list of keys.
+	AddOptions           map[string][]string // Add the options per given key.
+	Override             bool                // Override current tags when adding tags.
+	SkipUnexportedFields bool                // Skip unexported fields
+
+	Transform   string // Adds a transform rule when adding tags. Current options: [snakecase, camelcase, lispcase, pascalcase, titlecase, keep].
+	Sort        bool   // Sort tags in increasing order according to the key name.
+	ValueFormat string // Format the given tag's value. i.e: \"column:{field}\", \"field_name={field}\"
+	Clear       bool   // Clear all tags
+	ClearOption bool   // Clear all tag options
+}
+
+// Apply applies the struct tag modifications of the receiver to all
+// structs contained within the given node between start and end positions, mutating its input.
+func (m *Mutation) Apply(fset *token.FileSet, node ast.Node, start, end token.Pos) error {
+	err := m.Validate()
+	if err != nil {
+		return err
+	}
+	_, err = m.Rewrite(fset, node, fset.Position(start).Line, fset.Position(end).Line)
+	if err != nil {
+		if _, ok := err.(*RewriteErrors); !ok {
+			return err
+		}
+	}
+	return nil
+}
+
+func (m *Mutation) process(fieldName, tagVal string) (string, error) {
```

Comment:
How about calling this 'processField', with a docstring that explains that the return is the new struct tag value.